samza-core/src/main/java/org/apache/samza/application/StreamApplicationDescriptorImpl.java

```diff
@@ -112,7 +112,6 @@ public StreamApplicationDescriptor withDefaultSystem(SystemDescriptor<?> default
       return expander.get().apply(this, inputDescriptor);
```

Comment:
We probably should throw exception for ExpandingInputDescriptor calling withBroadcastPartitions().